core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java

```diff
@@ -56,6 +57,23 @@
 
 public class TestCatalogUtilDropTable extends HadoopTableTestBase {
 
+  @Test
+  public void dropTableDataDoNotThrowWhenReadOrDeletesFail() {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    table.newAppend().appendFile(FILE_B).commit();
+
+    TableMetadata tableMetadata = readMetadataVersion(3);
+
+    FileIO fileIO = Mockito.mock(FileIO.class);
+    Mockito.doThrow(new RuntimeException()).when(fileIO).deleteFile(ArgumentMatchers.anyString());
+    Mockito.doThrow(new RuntimeException()).when(fileIO).newInputFile(ArgumentMatchers.anyString());
```

Comment:
this is really the only mocking that this test requires, so I'd remove the other ones